src/cljdoc/server/pedestal.clj

```diff
@@ -152,20 +154,28 @@
      :enter (fn releases-loader-inner [ctx]
               (pu/ok ctx (versions-data searcher store route-name ctx)))})])
 
+(defn load-cache-bundle
+  "Given a `store` (see `:cljdoc/storage` in system.clj), `pom-info` (as returned from
+  `load-pom`), and a `version-entity` (see `:cljdoc.spec/version-entity` in spec.cljc),
+  load a `cache-bundle` from storage."
+  [store pom-info version-entity]
+  (let [params (assoc version-entity :dependency-version-entities (:dependencies pom-info))]
+    (log/info "Loading artifact cache bundle for" params)
+    (when (storage/exists? store params)
+      (storage/bundle-docs store params))))
+
```

Comment:
Yep! I have a nitpick in general with these sorts of things where I like handlers to be as simple as possible so that anything needed to pull together data can be re-used while keeping the handlers easy to understand.